### DISCUSSION TITLE
Support URL preservation for partial reload requests to API routes

### DISCUSSION THREAD

**Discussion by nathanmedz** [inertiajs/inertia#1689] [Ideas]
As far as I can tell, if you make a request to a url that is not your current url there is currently no way to prevent Inertia from modifying the browser url when you receive a response, even if your response renders the same component. With partial renders, this would mean that you can use API endpoints to fetch/ modify part of the data on your page without a full data reload. This is especially useful for pages with primary data that is expensive to fetch and secondary data that is frequently edited. 

Example - making a request from something like `https://mysite.com/home`, 
```
router.post(`/some_api_route`, data, {only: ['partial_data']})
```
this will partially reload the data as expected, but will also modify the browser url to `https://mysite.com/some_api_route` which is undesirable in many cases

**Comment by Tivixo** [comment]
Maybe this could help: https://github.com/inertiajs/inertia/pull/1009

But it doesn't work, I ask here about it: https://github.com/inertiajs/inertia/discussions/1702
Please tell me if you find out how to use it.
